src/js/utils/FormUtil.js

```diff
@@ -0,0 +1,89 @@
+function cloneDefinition(array) {
+  return array.map(function (object) {
+    if (Array.isArray(object)) {
+      return object.slice();
+    }
+
+    if (typeof object === 'object') {
+      return Object.assign({}, object);
+    }
+
+    return object;
+  });
+}
```

Comment:
We should consider moving this to a more generic file (`Util.js`? ). We may need a good solution for deep clones also in other sections. See https://github.com/mesosphere/marathon-ui/blob/master/src/js/helpers/Util.js#L132